lib/proxy/request.rb

```diff
@@ -26,15 +26,16 @@ def uri(path)
 
     def add_headers(req, headers={})
       req.add_field('Accept', 'application/json,version=2')
-      req.content_type = 'application/json'
+      req.content_type = headers["Content-Type"] || 'application/json'
       headers.each do |k, v|
         req.add_field(k, v)
       end
       req
     end
 
-    def create_post(path, body, headers={})
-      req = Net::HTTP::Post.new(uri(path).path)
+    def create_post(path, body, headers={}, query={})
+      uri = uri(path)
+      req = Net::HTTP::Post.new("#{uri.path || '/'}?#{query_string(query)}")
```

Comment:
If there is a query string, can we set this in the uri? Manually building query string just causes trouble in my experience.

```
uri.query = query_string(query)
```

Reply:
Sure.